rocky/reports/report_types/mail_report/report.py

```diff
@@ -0,0 +1,77 @@
+from dataclasses import dataclass
+from datetime import datetime
+from logging import getLogger
+from typing import Any, Dict
+
+from django.utils.translation import gettext_lazy as _
+
+from octopoes.models import Reference
+from octopoes.models.ooi.dns.zone import Hostname
+from octopoes.models.ooi.network import IPAddressV4, IPAddressV6
+from reports.report_types.definitions import Report
+
+logger = getLogger(__name__)
+
+
+@dataclass
+class System:
+    system_types: list
+    oois: list
+
+
+class MailReport(Report):
+    id = "mail-report"
+    name = _("Mail Report")
+    description = _("System specific mail report that focusses on IP addresses and hostnames.")
+    plugins = {"required": ["nmap"], "optional": []}
+    input_ooi_types = {Hostname, IPAddressV4, IPAddressV6}
+    template_path = "mail_report/report.html"
+
+    number_of_spf = 0
+    number_of_dmarc = 0
+    number_of_dkim = 0
+
+    def generate_data(self, input_ooi: str, valid_time: datetime) -> Dict[str, Any]:
+        reference = Reference.from_str(input_ooi)
+        hostnames = []
+        mail_security_measures = {}
+
+        if reference.class_type == Hostname:
+            hostnames = [reference]
+            measures = self.__get_measures(valid_time, hostnames)
+            mail_security_measures = {"hostname": hostnames[0].tokenized.name, "measures": measures}
+        elif reference.class_type in (IPAddressV4, IPAddressV6):
```

Comment:
Here you can simply use `IPAddress` since that's the base class
```suggestion
        elif reference.class_type == IPAddress:
```

Reply:
The base class is not equal to its subclass, so the suggested change won't work. I also don't think it's a problem to list both object types.